### PR TITLE
feat: 增加角头和序号列的交互能力

### LINKED ISSUE #2480: 🙏 透视表支持点击角头（列）字段时，选中相应行
### 🏷 Version
s2-react-v1.46.0-alpha.2

### Sheet Type
PivotSheet

### 希望透视表支持点击角头（列）字段时，选中相应行
image

### LINKED ISSUE #2563: 透视表树结构展开按钮不能正确的渲染
### 🏷 Version

| Package        | Version |
| -------------- | ------- |
| @antv/s2       |      next   |
| @antv/s2-react |         |
| @antv/s2-vue   |         |

### Sheet Type

- [X] PivotSheet
- [ ] TableSheet
- [ ] GridAnalysisSheet
- [ ] StrategySheet
- [ ] EditableSheet

### 🖋 Description
s2_table_index
1. 使用自定义layoutSeriesNumberNodes自定义序号列, 折叠按钮跑到了 序号单元格，图中1。
2. 点击第一个折叠按钮，会触发所有行展开，图中2

### PR DESCRIPTION
**Pull request by lijinke666**
### 👀 PR includes

✨ Feature

- [x] New feature

🐛 Bugfix

- [x] Solve the issue and close #2563 

### 📝 Description

1. 新增行角头的交互能力: 现在点击行角头, 会选中对应的列头. close #2480 

| Before | After |
| ------ | ----- |
| image| image |

2. 新增序号角头的交互能力: 现在点击序号角头, 选选中当前列

| Before | After |
| ------ | ----- |
| image | image |

3. 序号列支持 hover 高亮, 和其他单元格保持体验一致

| Before | After |
| ------ | ----- |
| image | image |

4. 数值单元格行列十字高亮, 现在会包含序号单元格

| Before | After |
| ------ | ----- |
| image | image |
|  image | image |

5. 修复透视表树状模式下, 全部展开收起 icon 渲染位置错误的问题 close #2563
 
| Before | After |
| ------ | ----- |
| image | image |

### 🔍 Self-Check before the merge

- [x] Add or update relevant docs.
- [x] Add or update relevant demos.
- [x] Add or update test case.
- [x] Add or update relevant TypeScript definitions.

### CODE REVIEW

File: packages/s2-core/src/interaction/root.ts [hunk 1]
```diff
@@ -629,4 +637,64 @@ export class RootInteraction {
...
+
+      const activeHeaderCells = [...activeSeriesNumberCells, ...activeRowCells];
+
+      forEach(activeHeaderCells, (cell) => {
+        cell.updateByState(stateName);
+      });
+    }
+  }
+
+  public updateDataCellRelevancyColCells(
```
Comment (wjgogogo) [inline packages/s2-core/src/interaction/root.ts:685]:
```suggestion
  public updateDataCellRelevantColCells(
```

Comment (lijinke666) [reply packages/s2-core/src/interaction/root.ts:685]: done

**Pull request merged**